### DISCUSSION TITLE
"Save As" Dialog

### DISCUSSION THREAD

**Discussion by kewartoscar** [zauberzeug/nicegui#2375] [Q&A] [answered]
### Question

Is there a way to use ui.upload or ui.download to create a "Save As" like dialog in NiceGui.
For instance, if I have a dataframe by name "df", which I want to convert to CSV. (Using the code df.to_csv("C:/my_preferred_file_name.csv")
I would want a dialog box with which I can choose a directory, and enter my preferred filename.

**Comment by falkoschindler** [comment]
I could imagine something like this:

```py
with ui.dialog() as dialog, ui.card():
    ui.label('Save as...')
    filename = ui.input('filename', value='file.txt')

    def save():
        ui.download(b'Hello world!', filename=filename.value)
        dialog.close()
    ui.button('Save', on_click=save)

ui.button('Open dialog', on_click=dialog.open)
```
<img width="502" alt="Screenshot 2024-01-13 at 21 00 20" src="https://github.com/zauberzeug/nicegui/assets/5767091/d82b84ca-2e08-44ec-8c8d-53439f8ec891">

**Reply by kewartoscar** [reply]
Thanks for your quick response. But where would would the df.to_csv('filename.value') be located?

**Comment by kewartoscar** [comment, marked as answer]
Finally, after several trial, using your code as guide, the code below did the magic.
```py
with ui.dialog() as dialog, ui.card():
    ui.label('Save as...')
    with ui.row():
        ui.label('File name:')
        filename = ui.input(' ', placeholder='Enter file name').props('outlined dense')

    def save():
        csv_file = dataframe_table.to_csv().encode()
        ui.download(src=csv_file, filename=f"{filename.value}.csv")
        dialog.close()
    ui.button('Save', on_click=save, icon='save')

ui.button('Open dialog', on_click=dialog.open)
```
